woocommerce/google-listings-and-ads
# Update UI to support minimum and maximum shipping times.

### Changes proposed in this Pull Request:



Part of pcTzPl-2qP-p2

This PR introduces a UI to manage minimum and maximum shipping times, and it updates the input to include increase and decrease buttons (like a stepper input). Additionally, we're considering showing "Same Day" instead of 0 to make it clearer for merchants.

![image](https://github.com/user-attachments/assets/a77918ce-f688-4aef-af4d-737cb5d3a9a5)


Since mixing a string like "Same Day" with the actual integer values for shipping times could make the code more complex, I opted to use the placeholder attribute to display "Same Day" instead of 0 days to keep things simple. However, I'm open to other suggestions or approaches.

For more context see this comment here: pcTzPl-2qP-p2#comment-3652

#### Other Steeper Inputs in WC

https://github.com/woocommerce/woocommerce/blob/64dcafe29b52cff2dee1d56f5e06caad09a0c470/packages/js/product-editor/src/components/number-control/number-control.tsx#L166-L222

### Screenshots:




### Detailed test instructions:


1. Follow the steps mentioned in this PR to update the database: https://github.com/woocommerce/google-listings-and-ads/pull/2520
2. Navigate to Dashboard -> Free Listings -> Edit
3. Test the new input buttons by increasing or decreasing the values. If the value is 0, it should display "Same Day".
4. Verify that the data is being stored correctly.
5. Add and modify shipping times as needed.
6. Disconnect all your accounts.
7. Do the onboarding, and see if everything works correctly.

### Additional details:

I haven't updated the add and edit modals to use the stepper component yet to make reviewing this PR easier. Also, I want to ask for confirmation from the designers to see if we want to implement it there as well.


### Changelog entry

>
